planner/optimize.go

```diff
@@ -621,12 +621,63 @@ func ExtractSelectAndNormalizeDigest(stmtNode ast.StmtNode, specifiledDB string)
 	return nil, "", "", nil
 }
 
+// ExtractSelectAndNormalizeDigestForBinding extract the select statement and normalize it with additional binding specific rules
+func ExtractSelectAndNormalizeDigestForBinding(stmtNode ast.StmtNode, specifiledDB string) (ast.StmtNode, string, string, error) {
```

Comment:
How about adding a flag in `ExtractSelectAndNormalizeDigest` like `ExtractSelectAndNormalizeDigest(..., forBinding bool)` instead of creating a new similar function?